### PR TITLE
NativeAOT-LLVM: add some support for floats/doubles/uint/long/byte/ubyte/short/ushort and GT_LE/LT/GE/GT

### PR DESCRIPTION
**Pull request by yowl**
This PR adds some support in the RyuJIT backend for more types and some binary arithmetic comparisons.  

Increases coverage by about 1%.

### CODE REVIEW

**Comment by yowl** [comment]
Hmm, it "Just worked".  I'll leave the debug in, in case it happens again.

File: src/coreclr/jit/llvm.cpp [hunk 1]
```diff
@@ -853,13 +883,34 @@ void buildCast(llvm::IRBuilder<>& builder, GenTreeCast* cast)
...
 }
 
+void buildCnsDouble(llvm::IRBuilder<>& builder, GenTree* node)
+{
+    if (node->gtType == var_types::TYP_DOUBLE)
+    {
+        mapGenTreeToValue(node, llvm::ConstantFP::get(Type::getDoubleTy(_llvmContext), node->AsDblCon()->gtDconVal));
+        return;
+    }
+    failFunctionCompilation(); // TODO-LLVM: how are floats and decimals done?
```
Comment (SingleAccretion) [inline src/coreclr/jit/llvm.cpp]: > how are floats and decimals done?

Floats are also stored in `GenTreeCnsDbl`, you just have to check for `TypeIs(TYP_FLOAT)` and downcast `gtDconVal`. Decimals do not exist at the IL level...

File: src/coreclr/jit/llvm.cpp [hunk 2]
```diff
@@ -853,13 +883,34 @@ void buildCast(llvm::IRBuilder<>& builder, GenTreeCast* cast)
...
+                : builder.CreateFPToSI(getGenTreeValue(cast->CastOp()).getValue(builder), getLLVMTypeForVarType(TYP_INT)));
+    }
     else
     {
         // TODO: other casts
         failFunctionCompilation();
     }
 }
 
+void buildCnsDouble(llvm::IRBuilder<>& builder, GenTree* node)
```
Comment (SingleAccretion) [inline src/coreclr/jit/llvm.cpp]: Use the strongly-typed `GenTreeCnsDbl*`?

File: src/coreclr/jit/llvm.cpp [hunk 3]
```diff
@@ -853,13 +883,34 @@ void buildCast(llvm::IRBuilder<>& builder, GenTreeCast* cast)
...
+{
+    if (node->gtType == var_types::TYP_DOUBLE)
```
Comment (SingleAccretion) [inline src/coreclr/jit/llvm.cpp]:
```suggestion
    if (node->TypeIs(TYP_DOUBLE))
```

File: src/coreclr/jit/llvm.cpp (hunk 2 above)
Comment (yowl) [reply src/coreclr/jit/llvm.cpp]: Thanks

File: src/coreclr/jit/llvm.cpp (hunk 1 above)
Comment (yowl) [reply src/coreclr/jit/llvm.cpp]: `ConstantFP::get` handles the downcast, so I left that out.  I did check and the new `TYP_FLOAT` path is exercised.

Comment (SingleAccretion) [reply src/coreclr/jit/llvm.cpp]: I think you can also remove the `failFunctionCompilation` at this point, and do something like `if (node->TypeIs(TYP_DOUBLE)) else { assert(node->TypeIs(TYP_FLOAT); }`, we will never see a constant of some other type here.

Comment (yowl) [reply src/coreclr/jit/llvm.cpp]: Thanks, done.

**Review by SingleAccretion — APPROVED** [review]

**Review by jkotas — APPROVED** [review]
Thanks!

**Pull request merged**